litex/soc/integration/builder.py

```diff
@@ -262,7 +262,7 @@ def _check_meson(self):
             msg = "Unable to find valid Meson build system, please install it with:\n"
             msg += "- pip3 install meson.\n"
             raise OSError(msg)
-        if (int(meson_version[0]) < meson_major_min) or (int(meson_version[1]) < meson_minor_min):
+        if (int(meson_version[0]) < meson_major_min) or (int(meson_version[0]) == meson_major_min and int(meson_version[1]) < meson_minor_min):
```

Comment:
See https://github.com/enjoy-digital/litex/issues/1545#issuecomment-1364736945

This check is still problematic and should be majorly refactored rather than hotfixed.